### PR TITLE
Added husky in the project

### LINKED ISSUE #534: Add husky to the project
### Description

currently in our project we don't have husky if we use husky we can easily formatting the code if before the make the commit by using pre commit 

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Ameerjafar**
## Changes Proposed

i have added the husky for the project with the help of this we don't worry to formatting the code. 
I just created the package.json file in the root folder because we need to add the .husky  folder and dependencies and then i have created the .husky folder inside the we have pre commit git hook file with the help of that we can able to format our code before the contributor make the commit.

## Check List

- [x] The title of my pull request is a short description of the changes
- [x] This PR relates to some issue: 
- [x] I have documented the changes made (if applicable)
- [ ] I have covered the changes with unit tests

### CODE REVIEW

**Comment by Ameerjafar** [comment]
@undera can you kindly review this pr sir.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (undera) [inline package.json:1]: All frontend-related stuff is contained under `/frontend` directory. You should incorporate your proposed changes there.

**Comment by undera** [comment]
I still don't understand why this needs to be part of the project. The change does not enforce anything on CI. If it's just local convenience helper - then it should remain for dev's won consideration.

File: package.json (hunk 1 above)
Comment (Ameerjafar) [reply package.json:1]: if I want to install husky in our project we should initialize the .husky folder where the .github folder is contained(root folder) at the same time we need to install dependencies that is the reason I have created the package.json file in the root folder sir @undera

**Comment by Ameerjafar** [comment]
> I still don't understand why this needs to be part of the project. The change does not enforce anything on CI. If it's just local convenience helper - then it should remain for dev's won consideration. We already have lint step for JS part in our CI. So I'm questioning if this PR adds any value, sorry.

yes sir husky will create some value for our project because if you have some constraints in the commit message we can use commit-msg hook, inside this file we can give the constraints if the contributor commit msg did not pass this constraints the git won't allow to make the commit and it gives so much more functionality for us.

**Comment by Ameerjafar** [comment]
@undera  yes sir without package.json file in the root folder we can do the husky stuff sir.

**Comment by undera** [comment]
I'm sorry, but I still don't understand which problem does this solve. Can you explain the problem it solves?
From your explanation, it sets some restriction on commit messages. The project does not have any problem with that, due to PR process by GitHub.

**Pull request closed without merging**